openstudiocore/src/model/GeneratorPVWatts.cpp

```diff
@@ -0,0 +1,443 @@
+    boost::optional<double> value = getDouble(OS_Generator_PVWattsFields::AzimuthAngle, true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool GeneratorPVWatts_Impl::isAzimuthAngleDefaulted() const {
+    return isEmpty(OS_Generator_PVWattsFields::AzimuthAngle);
+  }
+
+  boost::optional<PlanarSurface> GeneratorPVWatts_Impl::surface() const {
+    return getObject<ModelObject>().getModelObjectTarget<PlanarSurface>(OS_Generator_PVWattsFields::SurfaceName);
+  }
+
+  double GeneratorPVWatts_Impl::groundCoverageRatio() const {
+    boost::optional<double> value = getDouble(OS_Generator_PVWattsFields::GroundCoverageRatio, true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool GeneratorPVWatts_Impl::isGroundCoverageRatioDefaulted() const {
+    return isEmpty(OS_Generator_PVWattsFields::GroundCoverageRatio);
+  }
+
+  bool GeneratorPVWatts_Impl::setDCSystemCapacity(double dcSystemCapacity) {
+    bool result = setDouble(OS_Generator_PVWattsFields::DCSystemCapacity, dcSystemCapacity);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  bool GeneratorPVWatts_Impl::setModuleType(std::string moduleType) {
+    bool result = setString(OS_Generator_PVWattsFields::ModuleType, moduleType);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetModuleType() {
+    bool result = setString(OS_Generator_PVWattsFields::ModuleType, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setArrayType(std::string arrayType) {
+    bool result = setString(OS_Generator_PVWattsFields::ArrayType, arrayType);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetArrayType() {
+    bool result = setString(OS_Generator_PVWattsFields::ArrayType, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setSystemLosses(double systemLosses) {
+    bool result = setDouble(OS_Generator_PVWattsFields::SystemLosses, systemLosses);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetSystemLosses() {
+    bool result = setString(OS_Generator_PVWattsFields::SystemLosses, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setTiltAngle(double tiltAngle) {
+    bool result = setDouble(OS_Generator_PVWattsFields::TiltAngle, tiltAngle);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetTiltAngle() {
+    bool result = setString(OS_Generator_PVWattsFields::TiltAngle, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setAzimuthAngle(double azimuthAngle) {
+    bool result = setDouble(OS_Generator_PVWattsFields::AzimuthAngle, azimuthAngle);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetAzimuthAngle() {
+    bool result = setString(OS_Generator_PVWattsFields::AzimuthAngle, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setSurface(const PlanarSurface& surface) {
+    bool result = false;
```

Comment:
Should setting surface blank out tilt and azimuth?

Reply:
@macumber Wouldn't blanking them cause them to be set to their "default" values?